md4c/md4c.c

```diff
@@ -3497,6 +3535,74 @@ md_resolve_links(MD_CTX* ctx, const MD_LINE* lines, int n_lines)
     return 0;
 }
 
+static int
+md_is_inline_wikilink_spec(MD_CTX* ctx, int opener_index, int closer_index)
+
+{
+    MD_MARK* opener = &ctx->marks[opener_index];
+    MD_MARK* closer = &ctx->marks[closer_index];
+
+    int delim_index = opener_index;
+    MD_MARK* delim = &ctx->marks[delim_index];
+
+    /* No content; ignore. */
+    if (opener->end == closer->beg)
+        goto not_found;
+
+    while(delim_index < closer_index) {
+
+        /* The wiki link is ignored if the `|` delimiter is used, but without a
+         * label. TODO Unsure if this is the desired behavior. How about
+         * removing the `|` from the output instead, and keeping the link? */
+
+        /* The wiki link is also ignored if the delimiter is used, but without
+         * a target. */
+
+        if(delim->ch == '|' && (delim->end == closer->beg || delim->beg == opener->end)) {
+            goto not_found;
+        } else if(delim->ch == '|') {
+            opener->end = delim->beg;
+            break;
+        }
+
+        delim_index++;
+        delim = &ctx->marks[delim_index];
+    }
+
+    OFF off = closer->end;
+    int count = 0;
+    int newlines = 0;
+    int has_label = (opener->end - opener->beg > 2);
+
+    /* Check for newlines. */
```

Comment:
I do not see a point in this at all. The block parsing (the1 st parsing big step) already guarantees we do not try to match `[[` with `]]` across blocks (i.e. there is no blank line for example), and I do not see a principal reason why 
```
[[a long text
over multiple
lines|ID]]
```
should be rejected.

So imho we can still happily only scan over the ID part.

Also your approach would not be reliable:
1. `"\r\n"` is considered one new line digraph in Markdown.
2. Once you do not simply fail when reaching a new line, you can happily scan over contents which does not belong to the block contents like block decorations or indentation, breaking the logic with the text length limit.

    Consider e.g. a multi-line wiki-link used in a nested blockquote:
    ```
    > > > > [[foo
    > > > > bar|id]
    ```